merqube_client_lib/session.py

```diff
@@ -236,24 +228,34 @@ def __init__(
         """can specify your own oauth token,default is no token (only public APIs)"""
         super().__init__(token=token, **kwargs)
 
-    def handle_nonrecoverable(self, res: Response, e: Exception) -> None:
+    def handle_nonrecoverable(self, res: Response, exc: Exception, req_id: str) -> None:
         """helper function that handles a non recoverable error"""
-        logger.exception(e)
+        logger.exception(exc)
         try:
             # we may not have a json depending on the response
             rj = res.json()
         except (AttributeError, json.decoder.JSONDecodeError):
             rj = {}
-        logger.debug(f"Request failed with status {res.status_code}: {rj}")
-        raise APIError(code=res.status_code, response_json=rj)
+        logger.debug(f"Request failed with status {res.status_code}: {rj}. Request ID: {req_id}")
+        raise APIError(code=res.status_code, response_json=rj, request_id=req_id)
 
     def request_raise(self, method: httpm, url: str, **kwargs: Any) -> Response:
         """request method that logs the status code and raises on non 2XX"""
-        res = self.request(method, url, **kwargs)
+
+        # Generate a new requestid if this call isnt being made in a chain that already has it.
```

Comment:
moved from above since this header is a Merqube level-of-abstraction-var, not a lower level base session var